agent/agent.go

```diff
@@ -485,6 +461,49 @@ func (r *agent) recycleLoop(ctx context.Context) {
 	}
 }
 
+// subscriberLoop periodically updates the set of subscribers to this agent's
+// event queue.
+func (r *agent) subscriberLoop(ctx context.Context) {
+	ticker := r.Clock.NewTicker(subscriberInterval)
+	defer ticker.Stop()
+	for range ticker.Chan() {
+		if utils.IsContextDone(ctx) {
+			log.Info("Subscriber loop is stopping.")
+			return
+		}
+		if err := r.subscriberUpdate(); err != nil {
+			log.WithError(err).Warnf("Failed to update subscribers.")
+			continue
+		}
+	}
+}
+
+// subscriberUpdate updates the current set of subscribers. If any new master
+// nodes have joined the cluster, they will be subscribed to this agent's event
+// queue.
+func (r *agent) subscriberUpdate() error {
+	members, err := r.SerfClient.Members()
```

Comment:
Thinking out loud - I'm now wondering whether using serf separately from the satellite agent makes sense. If we embed it:
  * we would not need the client
  * we would not need an extra service in planet container
  * this could use event delegate for memberlist updates directly so you're not polling but receiving updates when they actually happen
  * lots of other benefits?...

Just something to think about.

Reply:
If we only interface with serf inside of satellite this would be a good idea. I'll take a look and try to work this change into the next refactoring.